include/circt/Conversion/StandardToHandshake.h

```diff
@@ -86,16 +123,35 @@ class HandshakeLowering {
       operands.push_back(startCtrl);
       rewriter.replaceOpWithNewOp<handshake::ReturnOp>(retOp, operands);
     }
+
+    // Store the number of block arguments in each block
+    DenseMap<Block *, unsigned> numArgsPerBlock;
+    for (auto &block : r.getBlocks())
+      numArgsPerBlock[&block] = block.getNumArguments();
+
+    // Apply SSA maximization on the newly added entry block argument to
+    // propagate it explicitly between the start-point of the control-only
+    // network and the function's terminators
+    if (failed(maximizeSSA(startCtrl, rewriter)))
```

Comment:
Would it be possible to apply the maximizeSSA pass before the `HandshakeLowering` class is kicked off? At some point, we would like to purge this class and everything we can already move out now would be a step into that direction.

Reply:
It should be possible, I will investigate that.